File: app/src/main/java/com/hfad/myferma/incubator/IncubatorMenuFragment.java
package com.hfad.myferma.incubator;


import android.app.FragmentTransaction;
import android.content.Intent;
import android.database.Cursor;
import android.os.Bundle;


import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import androidx.viewpager2.widget.ViewPager2;

import com.google.android.material.appbar.MaterialToolbar;
import com.google.android.material.floatingactionbutton.ExtendedFloatingActionButton;
import com.google.android.material.tabs.TabLayout;
import com.hfad.myferma.R;
import com.hfad.myferma.SalePackage.CustomAdapterSale;
import com.hfad.myferma.SalePackage.SaleActivity;
import com.hfad.myferma.WriteOff.WriteOffActivity;
import com.hfad.myferma.db.MyFermaDatabaseHelper;

import java.util.ArrayList;

public class IncubatorMenuFragment extends Fragment{
    private MyFermaDatabaseHelper myDB;
    private RecyclerView recyclerView;
    private ArrayList<String> id, name, type, data;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View layout = inflater.inflate(R.layout.fragment_incubator_menu, container, false);
//убириаем фаб кнопку
        ExtendedFloatingActionButton fab = (ExtendedFloatingActionButton) getActivity().findViewById(R.id.extended_fab);
        fab.setVisibility(View.GONE);

        //настройка верхнего меню фаб кнопку
        MaterialToolbar appBar = getActivity().findViewById(R.id.topAppBar);
        appBar.setTitle("Мои Инкубатор");
        appBar.setNavigationIcon(R.drawable.baseline_arrow_back_24);
        appBar.getMenu().findItem(R.id.delete).setVisible(true);
        appBar.setNavigationOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                getActivity().getSupportFragmentManager().popBackStack();
            }
        });

        TabLayout tabLayout = layout.findViewById(R.id.tab);
        ViewPager2 viewPager2 = layout.findViewById(R.id.view_pager);
        MenuAdapter menuAdapter = new MenuAdapter(getActivity());
        viewPager2.setAdapter(menuAdapter);

        tabLayout.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {
                viewPager2.setCurrentItem(tab.getPosition());
            }

            @Override
            public void onTabUnselected(TabLayout.Tab tab) {

            }

            @Override
            public void onTabReselected(TabLayout.Tab tab) {

            }
        });

        viewPager2.registerOnPageChangeCallback(new ViewPager2.OnPageChangeCallback() {
            @Override
            public void onPageSelected(int position) {
                super.onPageSelected(position);
                tabLayout.getTabAt(position).select();
            }
        });

        return layout;
    }

}

File: app/src/main/java/com/hfad/myferma/incubator/MenuIncubator/IncubatorMenuFragment.kt
package com.hfad.myferma.incubator.MenuIncubators

import android.os.Bundle
import android.view.LayoutInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.widget.Toolbar
import androidx.fragment.app.Fragment
import androidx.viewpager2.widget.ViewPager2
import androidx.viewpager2.widget.ViewPager2.OnPageChangeCallback
import com.google.android.material.appbar.MaterialToolbar
import com.google.android.material.dialog.MaterialAlertDialogBuilder
import com.google.android.material.floatingactionbutton.ExtendedFloatingActionButton
import com.google.android.material.tabs.TabLayout
import com.google.android.material.tabs.TabLayout.OnTabSelectedListener
import com.hfad.myferma.R
import com.hfad.myferma.WarehouseFragment
import com.hfad.myferma.db.MyFermaDatabaseHelper


class IncubatorMenuFragment : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        val layout: View = inflater.inflate(R.layout.fragment_incubator_menu, container, false)



        //убириаем фаб кнопку
        val fab: ExtendedFloatingActionButton =
            requireActivity().findViewById<View>(R.id.extended_fab) as ExtendedFloatingActionButton
        fab.visibility = View.GONE

        //настройка верхнего меню фаб кнопку
        val appBar: MaterialToolbar = requireActivity().findViewById(R.id.topAppBar)
        appBar.title = "Мои Инкубатор"
        appBar.setNavigationIcon(R.drawable.baseline_arrow_back_24)
        appBar.menu.findItem(R.id.delete).isVisible = true
        appBar.setOnMenuItemClickListener(Toolbar.OnMenuItemClickListener { item: MenuItem ->
            when (item.itemId) {
                R.id.delete ->deleteAllIcubator()
            }
            true
        })
        appBar.setNavigationOnClickListener {moveToNextFragment(WarehouseFragment())}

        val tabLayout: TabLayout = layout.findViewById(R.id.tab)
        val viewPager2: ViewPager2 = layout.findViewById(R.id.view_pager)

        val menuAdapter = MenuAdapter(requireActivity())

        viewPager2.adapter = menuAdapter
        tabLayout.addOnTabSelectedListener(object : OnTabSelectedListener {
            override fun onTabSelected(tab: TabLayout.Tab) {
                viewPager2.currentItem = tab.position
            }

            override fun onTabUnselected(tab: TabLayout.Tab?) {}
            override fun onTabReselected(tab: TabLayout.Tab?) {}
        })

        viewPager2.registerOnPageChangeCallback(object : OnPageChangeCallback() {
            override fun onPageSelected(position: Int) {
                super.onPageSelected(position)
                tabLayout.getTabAt(position)!!.select()
            }
        })
        return layout
    }


    private fun deleteAllIcubator() {
        val builder = MaterialAlertDialogBuilder(requireContext())
        builder.setTitle("Удалить все инкубаторы?")
        builder.setMessage("Вы уверены, что хотите удалить все инкубаторы включая архив?")
        builder.setPositiveButton(
            "Да"
        ) { dialogInterface, i ->
            val myDB = MyFermaDatabaseHelper(requireContext())
            myDB.deleteAllIncubator()

            val incubatorMenuFragment: IncubatorMenuFragment = IncubatorMenuFragment()
            requireActivity().supportFragmentManager.beginTransaction()
                .replace(R.id.conteiner, incubatorMenuFragment, "visible_fragment")
                .addToBackStack(null)
                .commit()
        }
        builder.setNegativeButton(
            "Нет"
        ) { dialogInterface, i -> }
        builder.create().show()
    }
    private fun moveToNextFragment(fragment: Fragment) {
        requireActivity().supportFragmentManager.beginTransaction()
            .replace(R.id.conteiner, fragment, "visible_fragment")
            .addToBackStack(null)
            .commit()
    }

}